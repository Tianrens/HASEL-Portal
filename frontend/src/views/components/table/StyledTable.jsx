import {
    TableContainer,
    Table,
    TableBody,
    TableHead,
    TableRow,
    TableCell,
    Paper,
} from '@mui/material';
import { React } from 'react';
import { headerCell, table } from './StyledTable.module.scss';

const HeaderTableCell = ({ children }) => <TableCell className={headerCell}>{children}</TableCell>;

const columnsCreator = (columns) => {
    // First column not aligned right
    const firstColumn = columns[0];
    const restColumns = columns.slice(1);

    return (
        <>
            <HeaderTableCell>{firstColumn}</HeaderTableCell>
            {restColumns.map((column) => (
                <HeaderTableCell key={column}>{column}</HeaderTableCell>
            ))}
        </>
    );
};

const StyledTable = ({ rows, columns, rowFactory }) => (
    <TableContainer component={Paper}>
        <Table className={table}>
            <TableHead>
                <TableRow>{columnsCreator(columns)}</TableRow>
            </TableHead>
            <TableBody>{rows.map((row) => rowFactory(row))}</TableBody>
        </Table>
    </TableContainer>
);

export default StyledTable;
