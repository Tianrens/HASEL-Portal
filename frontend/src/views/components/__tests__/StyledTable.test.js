import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider, TableRow, TableCell } from '@material-ui/core';
import StyledTable from '../table/StyledTable';

const columns = ['test'];
const rows = [{ title: 'test' }, { title: 'test2' }];
const rowFactory = (row) => {
    <TableRow key={row.title}>
        <TableCell component='th' scope='row'>
            {row.title}
        </TableCell>
    </TableRow>;
};

test('Styled Table renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <StyledTable columns={columns} rows={rows} rowFactory={rowFactory} />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
