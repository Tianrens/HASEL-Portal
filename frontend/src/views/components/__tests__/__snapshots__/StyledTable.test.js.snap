// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Table renders properly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
>
  <table
    className="MuiTable-root table css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          test
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
